### ISSUE TITLE
parallel computations may ignore conditions

### ISSUE THREAD

**Issue by AlexanderKalistratov** [numba/numba#5031]
## Reporting a bug

- [ ] I am using the latest released version of Numba (most recent is visible in
 the change log (https://github.com/numba/numba/blob/master/CHANGE_LOG).
- [x] I have included below a minimal working reproducer (if you are unsure how
 to write one see http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports).

```python
import numpy
import numba

def test_impl(arr, half=False):
    size = len(arr)
    parr = arr[0:size]

    if half:
        parr = arr[0:size//2]

    return parr.sum()

jtest = numba.njit(test_impl)
ptest = numba.njit(test_impl, parallel=True)

arr = numpy.ones(128)
print('sequential sum: ', jtest(arr, True))
print('parallel sum: ', ptest(arr, True))
```

Output:
```
sequential sum:  64.0
parallel sum:  128.0
```

Please note, this version works fine:
```python
import numpy
import numba

def test_impl(arr, half=False):
    size = len(arr)

    if half:
        parr = arr[0:size//2]
    else:
        parr = arr[0:size]

    return parr.sum()

jtest = numba.njit(test_impl)
ptest = numba.njit(test_impl, parallel=True)

arr = numpy.ones(128)
print('sequential sum: ', jtest(arr, True))
print('parallel sum: ', ptest(arr, True))
```

Output:
```
sequential sum:  64.0
parallel sum:  64.0
```

numba version: 6d7cddb9084d843e12bcdf9c443bc9cce56aa02f

**Comment by stuartarchibald** [comment]
Thanks for the report. This is a bug, it exits on 0.45.0, 0.46.0 series too (haven't checked further back).

**Comment by stuartarchibald** [comment]
CC @DrTodd13

**Comment by DrTodd13** [comment]
@sklam Close this issue?

**Comment by stuartarchibald** [comment]
Fixed in #5053, thanks @DrTodd13
